Replace Flask by Sanic (#113)
RasaHQ/rasa-sdk: CHANGELOG.rst

Added
-----

> - Add support for multiple sanic workers (configurable with the ``ACTION_SERVER_SANIC_WORKERS` environment variable)

Reviewer: ```suggestion
- Add support for multiple sanic workers (configurable with the ``ACTION_SERVER_SANIC_WORKERS`` environment variable)
```

Reviewer: Sorry, seems I missed some back ticks in the suggestion 🙈

Reviewer: I'll merge and add them quickly myself :-)

Reviewer: https://github.com/RasaHQ/rasa-sdk/pull/116

- Add support for async ``run`` methods in the ``Action`` class

> - Return status code ``404` in case requested action was not found

Reviewer: ```suggestion
- Return status code ``404`` in case requested action was not found
```

Added
-----

> - add unit tests of endpoints, actions and utils

Reviewer: We don't need a changelog update for the tests
```suggestion
```

> - add support of multiple sanic workers

Reviewer: ```suggestion
- Add support for multiple sanic workers (configurable with the ``ACTION_SERVER_SANIC_WORKERS` environment variable)
```

> - add support of async run method inside the action class

Reviewer: ```suggestion
- Add support for async ``run`` methods in the ``Action`` class
```

> - add typing for methods across the package

Reviewer: Same as for the tests (We appreciate both changes a lot 🚀 )
```suggestion
```

> - add handler of unknown action

Reviewer: Please also move this to `Changed` :-) 
```suggestion
- Return status code ``404` in case requested action was not found
```

> - add handler of empty body requests sent to webhook endpoint

Reviewer: Please also move this to ``Changed``
```suggestion
- Return status code ``400`` in case an empty request body was sent to the ``/webhook`` endpoint
```

Changed
-------

> - replace ``flask`` server framework by ``sanic``

Reviewer: ```suggestion
- Replace ``flask`` server framework with ``sanic``
```

> - replace ``flask_cors`` by ``sanic-cors``

Reviewer: ```suggestion
- Replace ``flask_cors`` with ``sanic-cors``
```

Removed
-------

> - remove ``flask`` framework and ``flask_cors``

Reviewer: I think that's already covered by the changelog above
```suggestion
```

> - remove ``requests`` package

Reviewer: ```suggestion
- Remove ``requests`` dependency
```

> - remove ``gevent`` package

Reviewer: ```suggestion
- Remove ``gevent`` dependency
```
